common/defs.go

```diff
@@ -82,6 +89,76 @@ type PortUpdateMessage struct {
 	Port uint64 `json:"port"`
 }
 
+type Endpoint struct {
+	CidrStr string `json:"cidr,omitempty"`
+	// TODO this can be collapsed into Cidr but needs
+	// work on JSON marshaller/unmarshaller to do that.
+	Cidr      net.IPNet `json:"-"`
+	TenantID  uint64    `json:"tenant_id,omitempty"`
+	TenantExternalID string  `json:"tenant_external_id,omitempty"`
+	SegmentID uint64    `json:"segment_id,omitempty"`
+	SegmentExternalID string `json:"segment_external_id,omitempty"`
+}
+
+const (
+	PolicyDirectionIngress = "ingress"
+	PolicyDirectionEgress  = "egress"
+)
+
+type PortRange [2]uint
+
+type Rule struct {
+	Protocol   string
+	Ports      []uint
+	PortRanges []PortRange
+	IcmpType   uint
+	IcmpCode   uint
+	Stateful   bool
+}
+
+type Rules []Rule
+
+// Metadata attached to entities for various external environments like openstack/kybernetes
+type Tag struct {
+	Key   string
+	Value string
+}
+
+type Policy struct {
+	Direction   string     `json:"direction,omitempty"`
+	Description string     `json:"description,omitempty"`
+	Name        string     `json:"name,omitempty"`
```

Comment:
Of course the limit is there. However, one of our key selling points is the operational simplicity when it comes to trouble shooting (for example). Seeing at least the first few characters of the policy name there is useful and helpful.

I'm not saying don't use the hex ID, but we can easily take the first couple of characters of the policy name and then add a hex ID to it. Here, take ROMANA and make ROM out of it, then use the first three or four letters of the policy name. Anything is better than nothing.